connector/ldap/ldap.go

```diff
@@ -456,6 +479,9 @@ func (c *ldapConnector) userEntry(conn *ldap.Conn, username string) (user ldap.E
 }
 
 func (c *ldapConnector) Login(ctx context.Context, s connector.Scopes, username, password string) (ident connector.Identity, validPass bool, err error) {
+	c.uid = username
+	c.pass = []byte(password)
```

Comment:
If I understand correctly what's happening here, this will set a struct variable for each login. That's probably not a great idea (for a number of reasons).

Username and password should instead be passed to a function. The do function probably needs some refactoring to do so.

Reply:
@sagikazarmark 
In simple-bind, the do() method requires username, password at bind time, so we need to pass them.

If passing them with struct ldapConnector is not appropriate, it should be passed as a method argument to the following method.

- Login()
- Refresh()  
- groups()
- do()

Is this arrangement correct?

Reply:
pushed test code.

Reply:
@sagikazarmark 
I changed to pass username and password to functions.